### ISSUE TITLE
NO clang-12 on debian

### ISSUE THREAD

**Issue by warjiang** [kcl-lang/kcl#1332]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
follow the instructions in [quick_start.md](https://github.com/kcl-lang/kcl/blob/main/docs/dev_guide/2.quick_start.md).

when install dependencies of clang & llvm, errors occured likes:
![image](https://github.com/kcl-lang/kcl/assets/6478745/edf73e30-b859-4820-b2c7-f2d63d0dfe0d)
The building system is debian 11.
After searching the pkg pages of debian,  `lld-12`  package does not exist under Debian system:
![image](https://github.com/kcl-lang/kcl/assets/6478745/24318b5b-96f6-4dca-bcff-5021e764fb28)

But `lld-12` can be found on ubuntu packages:
![image](https://github.com/kcl-lang/kcl/assets/6478745/a8a3cbe7-2003-4e68-b57b-b5f5e42bfe78)

### 2. What did you expect to see? (Required)
Seperate build instruction for debian and ubuntu if no maintainer for `lld-12` on debian system. Before build kcl on debian system, users maybe need to build llvm manually.

### 3. What did you see instead (Required)
no package for `lld-12` error
```
E: Package 'clang-12' has no installation candidate
E: Unable to locate package lld-12
```

### 4. What is your KCL components version? (Required)
branch `master`

**Comment by warjiang** [comment]
I found devcontainer inside this repo, with devcontainer, everything about environment is so easy, i found that the devcontainer.json missing one most important vscode extension: rust-analyzer, is it necessary to add it(currently it seems very useful for me)
![image](https://github.com/kcl-lang/kcl/assets/6478745/30ce51d2-5fc2-4e0f-8d41-e4eb3e0d9180)

**Comment by excalq** [comment]
Also trying to build KCL (as a locally installable release), in Ubuntu 24.04, and not having much success with `./scripts/build-llvm/build.sh 12.x` given many old C++ dependencies. Just curious how big task upgrading KCL's LLVM version is? Alternatively getting more detail in the docs about building without LLVM - though it looks like it's a requirement for building a usable "release"?

**Comment by warjiang** [comment]
> Also trying to build KCL (as a locally installable release), in Ubuntu 24.04, and not having much success with `./scripts/build-llvm/build.sh 12.x` given many old C++ dependencies. Just curious how big task upgrading KCL's LLVM version is? Alternatively getting more detail in the docs about building without LLVM - though it looks like it's a requirement for building a usable "release"?

@excalq  It's the problem of apt repo, I tested search by condition os=noble and keyword=llvm
https://packages.ubuntu.com/search?keywords=LLVM&searchon=names&suite=noble&section=all  , and no llvm-12 occurs.
You can search it with ubuntu22
https://packages.ubuntu.com/search?keywords=LLVM&searchon=names&suite=jammy&section=all
![image](https://github.com/kcl-lang/kcl/assets/6478745/a6bbd2d7-cde8-468e-a40e-a48dd6894174)



So  many problems with llvm and apt repo， i ithink you can try to build kcl in docker, the images `kcllang/kcl-builder:latest`  in https://github.com/kcl-lang/kcl/blob/main/.devcontainer/devcontainer.json may be useful.

**Comment by excalq** [comment]
Right, my next solution was the idea of building it in Docker, which I'll try next.
